src/main/java/tc/oc/pgm/api/map/factory/MapFactory.java

```diff
@@ -0,0 +1,28 @@
+package tc.oc.pgm.api.map.factory;
+
+import tc.oc.pgm.api.map.MapContext;
+import tc.oc.pgm.api.map.MapInfo;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.exception.MapException;
+import tc.oc.pgm.api.module.ModuleContext;
+import tc.oc.pgm.features.FeatureDefinitionContext;
+import tc.oc.pgm.filters.FilterParser;
+import tc.oc.pgm.kits.KitParser;
+import tc.oc.pgm.regions.RegionParser;
+import tc.oc.util.Version;
+
+/** A factory for creating {@link MapInfo}s and {@link MapContext}s. */
+public interface MapFactory extends ModuleContext<MapModule>, AutoCloseable {
+
+  RegionParser getRegions();
+
+  FilterParser getFilters();
+
+  KitParser getKits();
```

Comment:
We should deprecate these three parsers and explain why they shouldn't be here imho.

Reply:
It's too widely used at this point. But I'll add some descriptions.